rclcpp/test/rclcpp/node_interfaces/test_node_parameters.cpp

```diff
@@ -56,21 +70,15 @@ TEST(TestNodeParameters, list_parameters)
   auto list_result2 = node_parameters->list_parameters(prefixes, 1u);
   EXPECT_EQ(number_of_parameters + 1u, list_result2.names.size());
 
-  bool parameter_added = false;
-  for (const auto & name : list_result2.names) {
-    if (name.compare(parameter_name) == 0) {
-      parameter_added = true;
-      break;
-    }
-  }
-
-  EXPECT_TRUE(parameter_added);
-  rclcpp::shutdown();
+  EXPECT_NE(
+    std::find_if(
+      list_result2.names.begin(), list_result2.names.end(),
+      [parameter_name](const std::string & s) {return parameter_name.compare(s) == 0;}),
+    list_result2.names.end());
```

Comment:
@brawner this is correct, but FYI `std::find(list_result2.names.begin(), list_result2.names.end(), parameter_name)` should do too (since `std::basic_string<...>::operator==` uses `std::basic_string<...>::compare` [under the hood](https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp)).

Reply:
You're right, I was chasing a separate bug and got it confused with my choice of find vs find_if. Switched back to std::find.